### PR TITLE
fix errors on v1.6

### PR DESCRIPTION
**Pull request by aviatesk**
- xref: https://github.com/JuliaLang/julia/pull/36689

### CODE REVIEW

File: src/repl.jl [hunk 1]
```diff
@@ -47,7 +47,12 @@ function rewrite_with_ANSI(s, cursormove::Bool = false)
 
         l = textwidth(get_prompt(s))
         if !isa(s, LineEdit.SearchState)
-            LineEdit.write_prompt(termbuf, mode)
+            # xref: https://github.com/JuliaLang/julia/pull/36689
+            @static if VERSION ≥ v"1.6.0-DEV.531"
+                LineEdit.write_prompt(termbuf, mode, LineEdit.hascolor(terminal(s)))
```
Comment (aviatesk) [inline src/repl.jl:52]: Not sure `LineEdit.hascolor` stuff is necessary; maybe everyone who uses this package uses julia with `--color=yes`.

**Comment by aviatesk** [comment]
@KristofferC could you review this ?

**Comment by KristofferC** [comment]
Any idea why CI fails on master with this? Seems like it is not using the new definition?

**Comment by aviatesk** [comment]
My hunch is that `VERSION ≥ v"1.6.0-DEV.531"` was too new and so there are versions including JuliaLang/julia#36689 but before `DEV.531`. I eased the version check down to `v"1.6"` and let's see if CI passes.

**Comment by denius** [comment]
As I noticed the `VERSION` was `v"1.6.0-DEV.517"`, if it's matter. https://github.com/JuliaLang/julia/commit/a72cfb723101168766720e15ecda6b602ccdea5c#commitcomment-40845087

**Pull request merged**